### PR TITLE
Feature/create model

### PR DESCRIPTION
**Pull request by raodz**
## Description

1. **Added `BaldOrNotModel` Class**:
   - Introduced a new model class `BaldOrNotModel` which inherits from `tf.keras.Model`.
   - This class is designed for binary classification to determine if a person in an image is bald or not.

2. **New Files Related to Constants**:
   - Added `constants.py`: Contains various constants used throughout the project.
   - Added `config_constants.py`: Contains constants that are loaded from the `config.yaml` file.
   - Added `config.yaml`: This configuration file is used to define the constants loaded by `config_constants.py`.

3. **Removed `mypy` Package from `.pre-commit-config.yaml`**:
   - The `mypy` package was removed from the `.pre-commit-config.yaml` file.
   - This decision was made because `mypy` was not providing significant benefits and was starting to generate errors.

## Testing

The `BaldOrNotModel` has been thoroughly tested in two test files:
   - `test_create_model.py`: Tests the creation and structure of the model.
   - `test_untrained_model.py`: Tests the functionality and behavior of the untrained model.

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Tests update
- [x] Documentation update
- [x] Changes that do not directly affect the source code or tests

## Checklist
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by jakub1090cn — APPROVED** [review]

File: src/config_constants.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import os
+import yaml
+
+config_path = os.path.join(os.path.dirname(__file__), "..", "config.yaml")
+with open(config_path, "r") as file:
+    config = yaml.safe_load(file)
+
+DENSE_UNITS = config["dense_units"]
```
Comment (skrzypczykt) [inline src/config_constants.py:8]: To jest takie sobie, bo trzeba wszystkie elementy do zmiennej pakować. Prawdopodobnie możemy to zmienić w PRce dotyczącej configów, więc zostawmy.

File: src/model.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        `dropout_rate` is set.
+        predictions (tf.keras.layers.Layer): Dense layer with 1 unit and
+        sigmoid activation.
+    """
+
+    def __init__(
+        self, freeze_backbone: bool = True, dropout_rate: float | None = None
+    ):
+        super().__init__()
+        self.convnext_tiny: tf.keras.Model = (
```
Comment (skrzypczykt) [inline src/model.py:33]: zmienilbym nazwe zmiennej na self.backbone

File: src/model.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            DENSE_UNITS, activation="relu"
+        )
+        self.dropout: tf.keras.layers.Layer | None = (
+            tf.keras.layers.Dropout(dropout_rate)
+            if dropout_rate is not None
+            else None
+        )
+        self.predictions: tf.keras.layers.Layer = tf.keras.layers.Dense(
+            1, activation="sigmoid"
+        )
```
Comment (skrzypczykt) [inline src/model.py:54]: to chyba powinno być jakoś zgrupowane, może jedna zmienna `self.classifier` i tam wewnątrz już lista warstw, np. za pomocą `Sequential` ?

File: tests/test_create_model.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+def model() -> BaldOrNotModel:
+    """
+    Fixture for creating an instance of the BaldOrNotModel.
+
+    Returns:
+        BaldOrNotModel: An instance of the BaldOrNotModel class.
+    """
+    return BaldOrNotModel()
+
+
```
Comment (skrzypczykt) [inline tests/test_create_model.py:17]: zmieniłbym na `return BaldOrNotModel(freeze_backbone=True, dropout_rate = None)`

File: tests/test_untrained_model.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        The predictions have the correct shape (num_images, 1).
+        The predictions are within the range [0, 1].
+    """
+    num_images = 3
+    fake_data = tf.random.normal(
+        shape=(num_images, IMG_LEN, IMG_LEN, NUM_CHANNELS)
+    )
+    model.compile(
+        optimizer="adam", loss="binary_crossentropy", metrics=["accuracy"]
+    )
```
Comment (skrzypczykt) [inline tests/test_untrained_model.py:54]: A po co to?

**Pull request merged**